docs/RFD-JobWorkerService.md

```diff
@@ -0,0 +1,366 @@
+  }
+
+  message JobStatusResponse {
+    string  status = 1;
+    int32   exitCode = 2;
+    string  exitReason = 3;
+  } 
+
+  message OutputResponse {
+    bytes   content = 1;
+  }
+```
+
+### Client/CLI 
+
+User interact with API via [CLI (A command-line interface)](https://en.wikipedia.org/wiki/Command-line_interface). 
+Here is example below how user interact via CLI and expected output
+
+> _Following command line package: https://github.com/urfave/cli considered as option for CLI implementation._
+
+#### Starting new job
+
+```text
+    client \
+        --ca-cert <PATH_TO_CA_CERT> \
+        --client-cert <PATH_TO_CLIENT_CERT> \
+        --client-key <PATH_TO_CLIENT_KEY> \
+    start \
+        --cpu 0.5 \
+        --memory 500000 \
+        --io 1000000 \
+        --c $(which date)
+```
+
+expected output
+
+```text
+Jod:<UUID> stared. 
+```
+
+> _Note: cpu, memory and io settings are not guarantee that system would allocate exactly same CPU time and memory.
+> Based on system load process can have more or less of these settings._
+
+#### Query job status
+
+```text
+    client \
+        --ca-cert <PATH_TO_CA_CERT> \
+        --client-cert <PATH_TO_CLIENT_CERT> \
+        --client-key <PATH_TO_CLIENT_KEY> \
+    status \
+        --id <UUID>
+```
+
+expected output if started
+```text
+Jod:<UUID> has status: Started. ExitCode:-1, ExitReason:
+```
+
+expected output if completed
+```text
+Jod:<UUID> has status: Completed. ExitCode:0, ExitReason:
+```
+
+expected output if failed to run
+```text
+Jod:<UUID> has status: Stopped. ExitCode:-1, ExitReason: <ERROR DETAILS>
+```
+
+#### Streaming job output
+```text
+    client \
+        --ca-cert <PATH_TO_CA_CERT> \
+        --client-cert <PATH_TO_CLIENT_CERT> \
+        --client-key <PATH_TO_CLIENT_KEY> \  
+    stream \
+        --id <UUID>
+```
+
+expected output if completed
+```text
+Sat Aug 24 12:43:23 PDT 2024
+```
+
+The command will continue to print the output until the job completes or stopped.
+
+In the case that the job has completed, the command will print the output and then exit.
+
+#### Stopping a job
+
+```text
+    client \
+        --ca-cert <PATH_TO_CA_CERT> \
+        --client-cert <PATH_TO_CLIENT_CERT> \
+        --client-key <PATH_TO_CLIENT_KEY> \  
+    stop \
+        --id <UUID>
+```
+
+expected output if completed
+```text
+Jod:<UUID> has status: Stopped. ExitCode:0, ExitReason:
+```
+
+### Test Plan
+Create a few end-to-end tests to verify some behaviors like:
+
+#### Positive use cases  
+1. `Use Case #1:` Starting simple job, check job status and let the job complete.
+2. `Use Case #2:` Starting simple job, Stop job, check job status is Stopped.
+3. `Use Case #3:` Start long-running job, check status, stream job output, waiting for completions, check job status.
+4. `Use Case #4:` Start Job for with User1, check job status, try to request status/stop job from User2. Expected behavior User2 should receive error. Stop job with User1 credentials.
```

Comment:
nit: `Expected behavior User2 should receive error.` feels like a negative case?

Reply:
No problem. Moved under `negative use cases`